### ISSUE TITLE
Correct code example for resolving risky-file-permissions does not pass lint

### ISSUE THREAD

**Issue by jamescooke** [ansible/ansible-lint#2988]
##### Summary

[Correct code example 2 for `risky-file-permissions`](https://ansible-lint.readthedocs.io/rules/risky-file-permissions/#correct-code) does not pass ansible-lint - raises `yaml[key-duplicates]`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint --version
ansible-lint 6.12.1 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console (paste below)
# Make a dir
mkdir -p roles/repro/tasks
# Write the correct code to main.yaml
cat > roles/repro/tasks/main.yaml
```

```yaml
---
- name: Safe example of using ini_file (1st solution)
  community.general.ini_file:
    path: foo
    create: false  # prevents creating a file with potentially insecure permissions
    mode: preserve

- name: Safe example of using ini_file (2nd solution)
  community.general.ini_file:
    path: foo
    mode: 0600  # explicitly sets the desired permissions, to make the results predictable
    mode: preserve
```

```sh
# Lint roles
ansible-lint roles/
```

##### Desired Behavior

Linting passes with 0 errors - this is the "correct" code.

Linting passes with no warnings, especially not warnings regarding the `risky-file-permissions` error type that we're reading the documentation trying to fix.

##### Actual Behavior

Linting fails with 1 error (and two warnings which are also concerning)

Error:
```
yaml[key-duplicates]: Duplication of key "mode" in mapping
roles/repro/tasks/main.yaml:12
```

Full output:
```
[WARNING]: While constructing a mapping from <unicode string>, line 10, column 5, found a duplicate dict key (mode). Using last defined value
only.
[WARNING]: While constructing a mapping from /tmp/alintbug/roles/repro/tasks/main.yaml, line 10, column 5, found a duplicate dict key (mode).
Using last defined value only.
WARNING  Unable to load module community.general.ini_file at roles/repro/tasks/main.yaml:2 for options validation
WARNING  Unable to load module community.general.ini_file at roles/repro/tasks/main.yaml:8 for options validation
WARNING  Unable to resolve FQCN for module community.general.ini_file
WARNING  Listing 3 violation(s) that are fatal
risky-file-permissions: File permissions unset or incorrect. (warning)
roles/repro/tasks/main.yaml:2 Task/Handler: Safe example of using ini_file (1st solution)

risky-file-permissions: File permissions unset or incorrect. (warning)
roles/repro/tasks/main.yaml:8 Task/Handler: Safe example of using ini_file (2nd solution)

yaml[key-duplicates]: Duplication of key "mode" in mapping
roles/repro/tasks/main.yaml:12

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental
  - yaml[key-duplicates]  # Violations reported by yamllint.

                            Rule Violation Summary                             
 count tag                    profile rule associated tags                     
     1 yaml[key-duplicates]   basic   formatting, yaml                         
     2 risky-file-permissions safety  unpredictability, experimental (warning) 

Failed after min profile: 1 failure(s), 2 warning(s) on 2 files.
```
